schema/schema.go

```diff
@@ -0,0 +1,416 @@
+	"path"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/tigrisdata/tigrisdb-client-go/driver"
+)
+
+const (
+	tagName       = "tigrisdb"
+	tagPrimaryKey = "primary_key"
+	tagPII        = "pii"
+	tagEncrypted  = "encrypted"
+	tagSkip       = "-"
+
+	typeInteger = "integer"
+	typeString  = "string"
+	typeBoolean = "boolean"
+	typeNumber  = "number"
+	typeArray   = "array"
+	typeObject  = "object"
+
+	formatInt32    = "int32"
+	formatByte     = "byte"
+	formatDateTime = "date-time"
+)
+
+type Field struct {
+	Name   string   `json:"title,omitempty"`
+	Type   string   `json:"type,omitempty"`
+	Format string   `json:"format,omitempty"`
+	Tags   []string `json:"tags,omitempty"`
+	Desc   string   `json:"description,omitempty"`
+	Fields []Field  `json:"properties,omitempty"`
+	Items  *Field   `json:"items,omitempty"`
+}
+
+type Schema struct {
+	Name       string   `json:"title,omitempty"`
+	Desc       string   `json:"description,omitempty"`
+	Fields     []Field  `json:"properties,omitempty"`
+	PrimaryKey []string `json:"primary_key,omitempty"`
+}
+
+// ModelName returns name of the model
+func ModelName(s interface{}) string {
+	t := reflect.TypeOf(s)
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return t.Name()
+}
+
+// TODO: Handle UUID, Datetime, Set types
+// Uint is translated to int64 to avoid integer overflow
+// Uint64 cannot be represented by the JSON schema types
+func translateType(t reflect.Type) (string, string, error) {
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	switch t.Kind() {
+	case reflect.Struct:
+		if t.PkgPath() == "time" && t.Name() == "Time" {
+			return typeString, formatDateTime, nil
+		}
+		return typeObject, "", nil
+	case reflect.Array:
+		if t.Elem().Kind() == reflect.Uint8 {
+			return typeString, formatByte, nil
+		}
+		return typeArray, "", nil
+	case reflect.Slice:
+		if t.Elem().Kind() == reflect.Uint8 {
+			return typeString, formatByte, nil
+		}
+		return typeArray, "", nil
+	case reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int:
+		return typeInteger, formatInt32, nil
+	case reflect.Uint8, reflect.Uint16:
+		return typeInteger, formatInt32, nil
+	case reflect.Int64, reflect.Uint, reflect.Uint32:
+		return typeInteger, "", nil
+	case reflect.String:
+		return typeString, "", nil
+	case reflect.Float32, reflect.Float64:
+		return typeNumber, "", nil
+	case reflect.Bool:
+		return typeBoolean, "", nil
+	case reflect.Map:
+		return typeObject, "", nil
+	}
+
+	return "", "", fmt.Errorf("unsupported type: '%s'", t.Name())
+}
+
+func isPrimaryKeyType(tp string) bool {
+	return tp != typeArray && tp != typeNumber && tp != typeObject && tp != typeBoolean
+}
+
+// Build converts structured schema to driver schema
+func Build(sch *Schema) (driver.Schema, error) {
+	return json.Marshal(sch)
+}
+
+// Build converts structured schema to driver schema
+func (sch *Schema) Build() (driver.Schema, error) {
+	return Build(sch)
+}
+
+func parsePrimaryKeyIndex(tag string) (int, error) {
```

Comment:
I'd suggest adding a comment on the parse method to tell what is the expected input. 

also "parsePrimaryKey" should be sufficient.